cylc/flow/network/client.py

```diff
@@ -272,13 +277,30 @@ def _timeout_handler(workflow: str, host: str, port: Union[int, str]):
         """
         if workflow is None:
             return
+
+        try:
+            contact_data: Dict[str, str] = load_contact_file(workflow)
+        except (IOError, ValueError, ServiceFileError):
+            # Contact file does not exist or corrupted, workflow should be dead
+            return
+
+        contact_host: str = contact_data.get(ContactFileFields.HOST, '?')
+        contact_port: str = contact_data.get(ContactFileFields.PORT, '?')
+        if (
+            contact_host != host
+            or contact_port != str(port)
+        ):
+            raise CylcError(
+                f'The workflow is no longer running at {host}:{port}\n'
+                f'It has moved to {contact_host}:{contact_port}'
+            )
```

Comment:
This causes `cylc stop --poll` to fail if the target workflow got restarted quickly, hence failure of `tests/f/restart/00-pre-initial.t`.  I'll push a quick fix and merge the PR if it works.

Reply:
(The fix is to intercept CylcError in the poller, not to avoid raising it here).